install.sh

```
@@ -448,14 +452,118 @@ download_and_verify() {
+  local staging="$1" expected_version="$2"
+  local bin="$GENIE_HOME/bin" previous_dir="${GENIE_HOME}/bin/.previous"
+  local staged_bin="${staging}/genie" old_version backup entry base
+  mkdir -p "$previous_dir"
+
+  # Capture the currently-installed version BEFORE any sidecar (incl. VERSION)
+  # moves, so the backup filename reflects the binary actually being replaced.
+  if [[ -f "${bin}/genie" && ! -L "${bin}/genie" ]]; then
+    old_version="$(parse_version_token "$("${bin}/genie" --version 2>/dev/null || true)" 2>/dev/null || true)"
+    [[ -n "$old_version" ]] || old_version="previous"
+    backup="${previous_dir}/genie-${old_version}"
+    cp -p "${bin}/genie" "${backup}.staging.$$" &&
```

Shellcheck (0.11.0)

[info] Note that A && B || C is not if-then-else. C may run when A is true.

(SC2015)